File: TLBB1.5/Client/Assets/Client/Scripts/Table/Table_PvpnewReward.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_PvpnewReward : ITableOperate
{ private const string TAB_FILE_DATA = "pvpnew_reward.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_RANK_MAX,
ID_RANK_MIN,
ID_REWARD_1,
ID_REWARD_1_NUM,
ID_REWARD_2,
ID_REWARD_2_NUM,
ID_REWARD_3,
ID_REWARD_3_NUM,
ID_REWARD_4,
ID_REWARD_4_NUM,
ID_REWARD_5,
ID_REWARD_5_NUM,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private int m_RankMax;
 public int RankMax { get{ return m_RankMax;}}

private int m_RankMin;
 public int RankMin { get{ return m_RankMin;}}

private int[] m_Reward = new int[5];
 public int GetRewardbyIndex(int idx) {
 if(idx>=0 && idx<5) return m_Reward[idx];
 return -1;
 }

private int m_Reward1Num;
 public int Reward1Num { get{ return m_Reward1Num;}}

private int m_Reward2Num;
 public int Reward2Num { get{ return m_Reward2Num;}}

private int m_Reward3Num;
 public int Reward3Num { get{ return m_Reward3Num;}}

private int m_Reward4Num;
 public int Reward4Num { get{ return m_Reward4Num;}}

private int m_Reward5Num;
 public int Reward5Num { get{ return m_Reward5Num;}}

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_PvpnewReward _values = new Tab_PvpnewReward();
 _values.m_RankMax =  Convert.ToInt32(valuesList[(int)_ID.ID_RANK_MAX] as string);
_values.m_RankMin =  Convert.ToInt32(valuesList[(int)_ID.ID_RANK_MIN] as string);
_values.m_Reward [ 0 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_REWARD_1] as string);
_values.m_Reward [ 1 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_REWARD_2] as string);
_values.m_Reward [ 2 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_REWARD_3] as string);
_values.m_Reward [ 3 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_REWARD_4] as string);
_values.m_Reward [ 4 ] =  Convert.ToInt32(valuesList[(int)_ID.ID_REWARD_5] as string);
_values.m_Reward1Num =  Convert.ToInt32(valuesList[(int)_ID.ID_REWARD_1_NUM] as string);
_values.m_Reward2Num =  Convert.ToInt32(valuesList[(int)_ID.ID_REWARD_2_NUM] as string);
_values.m_Reward3Num =  Convert.ToInt32(valuesList[(int)_ID.ID_REWARD_3_NUM] as string);
_values.m_Reward4Num =  Convert.ToInt32(valuesList[(int)_ID.ID_REWARD_4_NUM] as string);
_values.m_Reward5Num =  Convert.ToInt32(valuesList[(int)_ID.ID_REWARD_5_NUM] as string);

 _hash.Add(nKey,_values); }


}
}

File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_PvpnewReward.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_PvpnewReward implements ITableOperate {
 public final static String TAB_FILE_DATA = "/pvpnew_reward.txt";
 public static final int ID_RANK_MAX = 0 ;
public static final int ID_RANK_MIN = 1 ;
public static final int ID_REWARD_1 = 2 ;
public static final int ID_REWARD_1_NUM = 3 ;
public static final int ID_REWARD_2 = 4 ;
public static final int ID_REWARD_2_NUM = 5 ;
public static final int ID_REWARD_3 = 6 ;
public static final int ID_REWARD_3_NUM = 7 ;
public static final int ID_REWARD_4 = 8 ;
public static final int ID_REWARD_4_NUM = 9 ;
public static final int ID_REWARD_5 = 10 ;
public static final int ID_REWARD_5_NUM = 11 ;
public static final int MAX_RECORD = 12 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private int m_RankMax;
 public int GetRankMax() { return m_RankMax;}

private int m_RankMin;
 public int GetRankMin() { return m_RankMin;}

private int m_Reward[] = new int[5];
 public int GetRewardbyIndex(int idx) {
 if(idx>=0 && idx<5) return m_Reward[idx];
 return -1;
 }

private int m_Reward1Num;
 public int GetReward1Num() { return m_Reward1Num;}

private int m_Reward2Num;
 public int GetReward2Num() { return m_Reward2Num;}

private int m_Reward3Num;
 public int GetReward3Num() { return m_Reward3Num;}

private int m_Reward4Num;
 public int GetReward4Num() { return m_Reward4Num;}

private int m_Reward5Num;
 public int GetReward5Num() { return m_Reward5Num;}

public boolean LoadTable(HashMap<Integer, Table_PvpnewReward> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_PvpnewReward> _hash = (HashMap<Integer, Table_PvpnewReward> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_PvpnewReward _values = new Table_PvpnewReward();
 _values.m_RankMax =  Integer.parseInt((String)valuesList.get(ID_RANK_MAX));
_values.m_RankMin =  Integer.parseInt((String)valuesList.get(ID_RANK_MIN));
_values.m_Reward [ 0 ] =  Integer.parseInt((String)valuesList.get(ID_REWARD_1));
_values.m_Reward [ 1 ] =  Integer.parseInt((String)valuesList.get(ID_REWARD_2));
_values.m_Reward [ 2 ] =  Integer.parseInt((String)valuesList.get(ID_REWARD_3));
_values.m_Reward [ 3 ] =  Integer.parseInt((String)valuesList.get(ID_REWARD_4));
_values.m_Reward [ 4 ] =  Integer.parseInt((String)valuesList.get(ID_REWARD_5));
_values.m_Reward1Num =  Integer.parseInt((String)valuesList.get(ID_REWARD_1_NUM));
_values.m_Reward2Num =  Integer.parseInt((String)valuesList.get(ID_REWARD_2_NUM));
_values.m_Reward3Num =  Integer.parseInt((String)valuesList.get(ID_REWARD_3_NUM));
_values.m_Reward4Num =  Integer.parseInt((String)valuesList.get(ID_REWARD_4_NUM));
_values.m_Reward5Num =  Integer.parseInt((String)valuesList.get(ID_REWARD_5_NUM));

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

